haystack/nodes/ranker/recentness_ranker.py

```diff
@@ -0,0 +1,159 @@
+import logging
+import warnings
+from collections import defaultdict
+from typing import List, Union, Optional, Dict, Literal
+
+from dateutil.parser import parse, ParserError
+
+from haystack.nodes.ranker.base import BaseRanker
+from haystack.schema import Document
+
+logger = logging.getLogger(__name__)
+
+
+class RecentnessRanker(BaseRanker):
+    outgoing_edges = 1
+
+    def __init__(
+        self,
+        date_identifier: str,
+        weight: float = 0.5,
+        top_k: Optional[int] = None,
+        method: Literal["reciprocal_rank_fusion", "score"] = "reciprocal_rank_fusion",
+    ):
+        """
+        This Node is used to rerank retrieved documents based on their age. Newer documents will rank higher.
+        The importance of recentness is parametrized through the weight parameter.
+
+        :param date_identifier: Identifier pointing to the date field in the metadata.
+                This is a required parameter, since we need dates for sorting.
+        :param weight: in range [0,1].
+                0 disables sorting by age.
+                0.5 content and age have the same impact
+                1 means sorting only by age, most recent comes first.
+        :param top_k: (optional) How many documents to return.
+                It can make sense to have large top-k values from the initial retrievers and filter docs down in the
+                RecentnessRanker with this top_k parameter.
+        :param method: Method used to combine retriever and recentness. Possible values are 'reciprocal_rank_fusion' (default) and 'score'
+                Make sure to use 'score' method only with retrievers/rankers that give back OK score in range [0,1]
```

Comment:
```suggestion
        :param method: Method used to combine retriever and recentness. Possible values are 'reciprocal_rank_fusion' (default) and 'score'.
                Make sure to use 'score' method only with retrievers/rankers that give back OK score in range [0,1].
```

Adding full stops here :)